core-java-modules/core-java-string-operations-2/src/test/java/com/baeldung/encode/StringEncodeUnitTest.java

```diff
@@ -0,0 +1,68 @@
+package com.baeldung.encode;
+
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.codec.binary.StringUtils;
+import org.junit.Test;
+
+public class StringEncodeUnitTest {
+
+    @Test
+    public void givenGermanAsciiString_whenComparing_thenCompareNotEquals() {
+        String germanString = "Entwickeln Sie mit Vergnügen";
+        byte[] germanBytes = germanString.getBytes();
+        String asciiEncodedString = new String(germanBytes, StandardCharsets.US_ASCII);
+
+        assertNotEquals(asciiEncodedString, germanString);
+    }
+
+    @Test
+    public void givenUsAsciiString_whenComparing_thenCompareNotEquals() {
+        String englishString = "Develop with pleasure";
+        byte[] englishBytes = englishString.getBytes();
+        String asciiEncondedEnglishString = new String(englishBytes, StandardCharsets.US_ASCII);
+
+        assertEquals(asciiEncondedEnglishString, englishString);
```

Comment:
swap order of arguments so that it matches the signature, assertEquals(expected, actual)

Reply:
Done